[tijngsheng] ip

tjingsheng opened the pull request:
# Duke project
## Hello from
```
      ██╗██╗███╗   ██╗ ██████╗     ███████╗██╗  ██╗███████╗███╗   ██╗ ██████╗ 
      ██║██║████╗  ██║██╔════╝     ██╔════╝██║  ██║██╔════╝████╗  ██║██╔════╝ 
      ██║██║██╔██╗ ██║██║  ███╗    ███████╗███████║█████╗  ██╔██╗ ██║██║  ███╗
 ██   ██║██║██║╚██╗██║██║   ██║    ╚════██║██╔══██║██╔══╝  ██║╚██╗██║██║   ██║
 ╚█████╔╝██║██║ ╚████║╚██████╔╝    ███████║██║  ██║███████╗██║ ╚████║╚██████╔╝
  ╚════╝ ╚═╝╚═╝  ╚═══╝ ╚═════╝     ╚══════╝╚═╝  ╚═╝╚══════╝╚═╝  ╚═══╝ ╚═════╝
```
## Project Description
This is my project, created from a template for a greenfield Java project.
It's named after the Java mascot _Duke_.
This project is a Java application that aims to provide a user-friendly task management system. It allows users to manage their tasks efficiently and keep track of their schedules.

## Features

- **Task Creation**: Users can add various types of tasks, including todo, deadline, and event tasks.
- **Task List**: View a list of all tasks, including their descriptions and completion status.
- **Task Completion**: Mark tasks as complete when they are done.
- **Task Deletion**: Remove tasks that are no longer needed.
- **Task Searching**: Search for tasks based on keywords.
- **Data Storage**: Data is automatically saved and loaded for seamless usage.

## Installation Option 1

1. Go to tjingsheng ip [releases](https://github.com/tjingsheng/ip/releases)
2. **Get Java**: Ensure you have a functioning version of Java.
3. **Copy the JAR File**: Copy the JAR file you want to run into an empty folder.
4. **Open a Command Window**: Open a command prompt (Windows) or terminal (macOS/Linux) in the same folder where you copied the JAR file.
5. **Run the Command**: Use the following command to run the JAR file:
6. **Enjoy**: Duke is very fun.

   ```shell
   java -jar duke.jar

## Installation Option 2

1. Clone the tjingsheng ip [repository](https://github.com/tjingshengg/ip) to your local machine.
2. Open the project in your preferred Java development environment.
3. Build and run the application.
4. Enjoy

## Todo List

- [x] Task 1 : Get duke
- [ ] Task 2 : Experiment with it
- [ ] Task 3 : Profit ???
- [ ] Task 4 : ~~Cry~~

## Project Sample
```
    /**
     * Runs the Duke application, allowing the user to interact with it via the command line.
     */
    public static void runDuke() {
        Scanner scanner = new Scanner(System.in);
        Ui.helloWorld();
        boolean isConversing = true;
        while (isConversing) {
            String rawCommand = Ui.getUserInput(scanner);
            TaskList taskList = TaskFileHandler.readFromFile();
            Ui.getBotMessage();
            Command command = Parser.parse(rawCommand);
            command.execute(taskList);
            TaskFileHandler.saveToFile(taskList);
            isConversing = !command.isByeCommand();
        }
    }
```

>>>>> "Duke hehe Duke hehe" - Jing Sheng

tjingsheng committed:
add-duke-level-0

tjingsheng committed:
add-duke-level-1

tjingsheng committed:
add-duke-level-2

tjingsheng committed:
add-duke-level-3

tjingsheng committed:
add-duke-level-4

tjingsheng committed:
add-duke-level-5

tjingsheng committed:
add-a-textuitesting

tjingsheng committed:
add-duke-level-6

tjingsheng committed:
add-a-enums

tjingsheng committed:
change-readme

tjingsheng committed:
Merge pull request #1 from tjingsheng/add-intro

change-readme

tjingsheng committed:
add-duke-level-7

tjingsheng committed:
Merge pull request #2 from tjingsheng/branch-Level-7

add-duke-level-7

tjingsheng committed:
add-duke-level-8

tjingsheng committed:
Merge pull request #3 from tjingsheng/branch-level-8

add-duke-level-8

tjingsheng committed:
add-a-moreoop

tjingsheng committed:
Merge pull request #4 from tjingsheng/a-more-oop

add-a-moreoop

tjingsheng committed:
add-a-gradle

tjingsheng committed:
Merge pull request #5 from tjingsheng/add-a-gradle

add-a-gradle

zhanyang01 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,30 @@
+    public static void runDuke() {
+        Scanner scanner = new Scanner(System.in);
+        Ui.helloWorld();
+        boolean converse = true;
```
boolean should sound like booleans, maybe isConverse is better?

tjingsheng replied:
good point

zhanyang01 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,30 @@
+import java.util.Scanner;
```
Nice! Imports are mostly very neat and it makes it readable! Good Job!

tjingsheng replied:
thank you

zhanyang01 commented on src/main/java/Parser/Parser.java:
```diff
@@ -0,0 +1,29 @@
+package Parser;
+
+import Command.*;
```
Should wildcard imports be used? Maybe it is better to list them out?

tjingsheng replied:
I think * is better as I am importing almost every command in that package

tjingsheng replied:
HMMM actually now that I saw this " Rationale: Importing classes explicitly gives an excellent documentation value for the class at hand and makes the class easier to comprehend and maintain. Appropriate tools should be used in order to always keep the import list minimal and up to date. IDE's can be configured to do this easily." in the style guide, I am feeling conflicted.

zhanyang01 replied:
How can IDE be configured to do it though?

zhanyang01 reviewed (commented):
I like how most of the coding standards and coding quality standards are met, good job for the IP so far!

tjingsheng committed:
add-a-junit

tjingsheng committed:
add-a-jar

tjingsheng committed:
Merge branch 'master' into add-a-jar

tjingsheng committed:
Merge pull request #11 from tjingsheng/add-a-jar

add-a-jar

tjingsheng committed:
add-a-javadoc

tjingsheng committed:
add-a-codingstandard

tjingsheng committed:
add-duke-level-9

tjingsheng committed:
Merge pull request #12 from tjingsheng/branch-A-JavaDoc

add-a-javadoc

tjingsheng committed:
Merge branch 'master' into branch-A-CodingStandard

tjingsheng committed:
Merge pull request #13 from tjingsheng/branch-A-CodingStandard

add-a-codingstandard

tjingsheng committed:
Merge branch 'master' into branch-Level-9

tjingsheng committed:
Merge pull request #14 from tjingsheng/branch-Level-9

add-duke-level-9

tjingsheng committed:
use-gfmd-in-README

rayshawntan commented on src/main/java/storage/TaskAdapter.java:
```diff
@@ -0,0 +1,49 @@
+package storage;
+
+import com.google.gson.*;
```
Should only import the classes needed here as mentioned in the coding standards

rayshawntan replied:
This as well!

rayshawntan commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,73 @@
+package parser;
+
+import command.*;
```
Should only import the classes needed here as mentioned in coding standards.

rayshawntan commented on src/main/java/storage/LocalDateAdapter.java:
```diff
@@ -0,0 +1,46 @@
+package storage;
+
+import com.google.gson.*;
```
Same here for this!

rayshawntan reviewed (commented):
Overall, I find your code easy to read especially with the javadocs that explains clearly what each class and methods are used for!

tjingsheng committed:
add-duke-a-checkstyle

tjingsheng committed:
Merge pull request #15 from tjingsheng/add-a-checkstyle

add-duke-a-checkstyle

tjingsheng committed:
Merge pull request #15 from tjingsheng/add-a-checkstyle

add-duke-a-checkstyle

tjingsheng committed:
Merge branch 'master' of https://github.com/tjingsheng/ip

tjingsheng committed:
add-duke-level-10

tjingsheng committed:
add-duke-level-10

tjingsheng committed:
Merge pull request #16 from tjingsheng/branch-Level-10

add-duke-level-10

tjingsheng committed:
add-missing-fonts

tjingsheng committed:
Merge branch 'master' into branch-Level-10

tjingsheng committed:
Merge pull request #17 from tjingsheng/branch-Level-10

add-duke-level-10

tjingsheng committed:
fix merge error

tjingsheng committed:
add-duke-level-10

tjingsheng committed:
Merge pull request #18 from tjingsheng/branch-Level-10

add-duke-level-10

tjingsheng committed:
add-a-varargs

tjingsheng committed:
Merge pull request #19 from tjingsheng/add-a-varags

add-a-varargs

tjingsheng committed:
add-a-assertions

tjingsheng committed:
add-a-codequality

tjingsheng committed:
Merge pull request #20 from tjingsheng/branch-A-CodeQuality

add-a-codequality

tjingsheng committed:
Merge branch 'master' into branch-A-Assertions

tjingsheng committed:
Merge pull request #21 from tjingsheng/branch-A-Assertions

add-a-assertions

tjingsheng committed:
Refactor based on Sharing iP code quality feedback

applied SLAP (and other abstraction mechanisms) to shorten methods e.g.,
extract some code blocks into separate methods. You may ignore this
suggestion if you think a longer method is justified in a particular
case.

Avoid committing binary files (e.g., *.class, *.jar, *.exe) or third-
party library files in to the repo. Removed Sono.zip.

Add improvements to exception handling.

tjingsheng committed:
add-bcd-extension

tjingsheng committed:
Merge pull request #23 from tjingsheng/add-bcd-extension

add-bcd-extension

tjingsheng committed:
remove redundant test in bye command

tjingsheng committed:
add-week-six

tjingsheng committed:
Merge pull request #24 from tjingsheng/add-week-six

add-week-six

tjingsheng committed:
fix-bugs

tjingsheng committed:
Merge pull request #25 from tjingsheng/fix-bug

fix-bugs

tjingsheng committed:
improve-application

tjingsheng committed:
Merge pull request #26 from tjingsheng/improve-application

improve-application

tjingsheng committed:
Update and rename README.md to docs/GFMDPractice.md

tjingsheng committed:
Update README.md

tjingsheng committed:
update-readme-ug

tjingsheng committed:
Merge pull request #27 from tjingsheng/update-readme-ug

update-readme-ug

tjingsheng committed:
fix-build-gradle
